mapswipe_workers/locust_files/load_testing.py

```diff
@@ -0,0 +1,132 @@
+# run with "locust -f locust_files/load_testing.py"
+# then set number of users and spawn rate in web interface
+# e.g. test with 100 users, and 15 users/sec
+# web interface: http://0.0.0.0:8089/
+
+import datetime
+import json
+import random
+from uuid import uuid4
+
+from locust import HttpUser, between, task
+
+from mapswipe_workers.definitions import logger
+from mapswipe_workers.utils import user_management
+
+
+class MapSwipeUser(HttpUser):
+    # assuming that users need between 30 sec and 120 sec to map a group
+    wait_time = between(30, 120)
+
+    def set_up_user(self):
+        # check if is already signed in
+        if self.signed_in_user is None:
+            logger.info("user is not signed in. Will create a new user.")
+            # create user if not exists
+            user = user_management.create_user(self.email, self.username, self.password)
+            self.user_id = user.uid
+
+            # sign in user
+            self.signed_in_user = user_management.sign_in_with_email_and_password(
+                self.email, self.password
+            )
+            logger.info("Created a new user.")
+        else:
+            logger.info("user is already signed in.")
+            pass
+
+    def create_mock_result(self, group):
+        """Create a result object for a build area project.
+
+        The result values are generated randomly.
+        """
+        start_time = datetime.datetime.utcnow().isoformat()[0:-3] + "Z"
+        end_time = datetime.datetime.utcnow().isoformat()[0:-3] + "Z"
+
+        x_min = int(group["xMin"])
+        x_max = int(group["xMax"])
+        y_min = int(group["yMin"])
+        y_max = int(group["yMax"])
+
+        results = {}
+        for x in range(x_min, x_max):
+            for y in range(y_min, y_max):
+                task_id = f"18-{x}-{y}"
+                result = random.choices([0, 1, 2, 3])[0]  # no, yes, maybe, bad_imagery
+                results[task_id] = result
+
+        data = {
+            "results": results,
+            "startTime": start_time,
+            "endTime": end_time,
+        }
+        return data
+
+    def set_firebase_db(self, path, data, token=None):
+        """Upload results to Firebase using REST api."""
+        request_ref = f"{path}.json?auth={token}"
+        headers = {"content-type": "application/json; charset=UTF-8"}
+        self.client.patch(
```

Comment:
this page explains it in more detail: https://firebase.google.com/docs/reference/rest/database

but you are right that `put` would be the right thing to use here. `patch` will also work.

`post` will not work since it is equivalent to `push` in Firebase.